lib/sort_by_length.rb

```diff
@@ -1,7 +1,25 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2)
+# Space complexity: O(n)
+# Source Citation: https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/04-cs-fundamentals/classroom/Sorting.md, Insertion Sort example implementation
+# Adjusted source citation example for modified problem
 def sort_by_length(my_sentence)
-  raise NotImplementedError, "Method not implemented"
+  word_array = my_sentence.split(" ")
```

Comment:
Because I made a mistake with naming the test file, you don't need to split the sentence as the argument is intended to be an array of strings to sort.